42.FragmentShaderPathTracer/common.glsl

```diff
@@ -678,10 +680,12 @@ bool closestHitProgram(in uint depth, in uint _sample, inout Ray_t ray, inout nb
 
 void main()
 {
+    const ivec2 imageExtents = nbl_glsl_swapchain_transform_preTransformExtents(cameraData.swapchainTransform, imageSize(outImage));
     const ivec2 coords = getCoordinates();
-    const vec2 texCoord = getTexCoords();
+    vec2 texCoord = (vec2(nbl_glsl_swapchain_transform_preTransform(cameraData.swapchainTransform, coords, imageSize(outImage))) / vec2(imageExtents));
+    texCoord.y = 1.0 - texCoord.y;
 
```

Comment:
Why do we need to flip y again if we did everything correctly?

Reply:
only one transform should exist

In an example that renders vertices to screen you apply it to the projection matrix (given triangle vertices you map them to screen). Remember rasterization is a scatter op.

However in an example such as 42, where you're casting rays, you're finding out worldspace positions for screenspace pixels so transform needs to be applied in reverse. Raycasting/post process is a gather op.

Reply:
@deprilula28 can you comment what @devshgraphicsprogramming wrote
Even in a comment above this line will do